src/Mod/Part/Gui/Command.cpp

```diff
@@ -1329,7 +1329,38 @@ CmdPartExtrude::CmdPartExtrude()
 void CmdPartExtrude::activated(int iMsg)
 {
     Q_UNUSED(iMsg);
-    Gui::Control().showDialog(new PartGui::TaskExtrusion());
+    /** If the user selects objects prior to executing the Extrude command
+     *  we can use those objects rather than opening the dialog
+     */
+
+    bool neverSkipDialog = App::GetApplication().GetUserParameter().GetGroup("BaseApp/Preferences/Mod/Part")->GetBool("NeverSkipDialog", false);

```

Comment:
You 4x almost the identical code. The only difference is the name of the Python function, the undo/redo text and the dialog class. To avoid the code duplication you can move this to a separate template function (the template parameter will be the name of the task dialog class, e.g. PartGui::TaskExtrusion) and two function arguments for the Python function and the text.